script/prerender/manage.ts

```diff
@@ -0,0 +1,397 @@
+        Id: batchIndex.toString(),
+        MessageBody: JSON.stringify({ payload: { page }, type: 'page' }),
+      })),
+      QueueUrl: workQueueUrl,
+    }));
+
+    const failedMessages = sendMessageBatchResult.Failed || [];
+    const numFailures = failedMessages.length;
+    if (numFailures > 0) {
+      throw new Error(`[SQS] [SendMessageBatch] Error: ${numFailures} out of ${pageBatch.length
+        } pages in a batch failed to be queued. Failures: ${JSON.stringify(failedMessages)}`);
+    }
+
+    const successfulMessages = sendMessageBatchResult.Successful || [];
+    const numSuccesses = successfulMessages.length;
+    if (numSuccesses < pageBatch.length) {
+      throw new Error(`[SQS] [SendMessageBatch] Unexpected response: received only ${numSuccesses
+        } successes out of ${pageBatch.length} pages in a batch but also no failures. Successes: ${
+        JSON.stringify(successfulMessages)}`);
+    }
+  }
+
+  console.log(`[${book.title}] All ${numBookPages} book pages queued`);
+
+  // console.log(`[${book.title}] Queuing sitemap`);
+
+  await sqsClient.send(new SendMessageCommand({
+    MessageBody: JSON.stringify({ payload: { pages, slug: book.slug }, type: 'sitemap' }),
+    QueueUrl: workQueueUrl,
+  }));
+
+  console.log(`[${book.title}] Sitemap queued`);
+
+  // Used in the sitemap index
+  return {
+    params: { book: { slug: book.slug } },
+    state: { bookUid: book.id, bookVersion: book.version },
+  };
+}
+
+async function manage() {
+  console.log('Preloading route components');
+
+  await Loadable.preloadAll();
+
+  archiveLoader = createArchiveLoader(REACT_APP_ARCHIVE_URL, {
+    appPrefix: '',
+    archivePrefix: ARCHIVE_URL,
+  });
+  osWebLoader = createOSWebLoader(`${OS_WEB_URL}${REACT_APP_OS_WEB_API_URL}`);
+
+  bookLoader = makeUnifiedBookLoader(archiveLoader, osWebLoader);
+
+  // We can start fetching book ToCs while the stack is created,
+  // but we need a limit so we don't use up all the memory
+  console.log(`Loading books in batches of ${MAX_CONCURRENT_BOOKS}`);
+
+  const allPagesQueuedPromise = asyncPool(
+    MAX_CONCURRENT_BOOKS, Object.entries(BOOKS), prepareAndQueueBook
+  );
+
+  // Wait for the workers stack to be ready
+  await waitUntilStackCreateComplete(
+    {client: cfnClient, maxWaitTime: WORKERS_DEPLOY_TIMEOUT_SECONDS, minDelay: 10, maxDelay: 10},
+    {StackName: WORKERS_STACK_NAME}
+  );
+
+  console.log('Retrieving queue URLs');
+
+  // Get the queue URLs
+  const describeStacksResult = await cfnClient.send(
+    new DescribeStacksCommand({StackName: WORKERS_STACK_NAME})
+  );
+
+  const stacks = assertDefined(
+    describeStacksResult.Stacks, '[CFN] [DescribeStacks] Unexpected response: missing Stacks key'
+  );
+  const stack = assertDefined(stacks[0], `${WORKERS_STACK_NAME} stack not found`);
+  const outputs = assertDefined(
+    stack.Outputs, '[CFN] [DescribeStacks] Unexpected response: missing stack Outputs key'
+  );
+
+  let workQueueUrl: string | undefined;
```

Comment:
i get that its less efficient to loop over the outputs twice but since there are only like two outputs i feel like it would be fine to do a `find` here and it would be much more readable

Reply:
Done, replaced with 2 calls to `find()`